polyfills/URL/detect.js

```diff
@@ -0,0 +1,25 @@
+(function (global) {
+	/*
+	 * Browsers may have:
+	 * No global URL object
+	 * URL with static methods only - may have a dummy constructor
+	 * URL with members except searchParams
+	 * Full URL API support
+	 */
+	'use strict';
+
+	try {
+		if (global.URL) {
+			var nativeURL = new global.URL('http://example.com');
+
+			if ('searchParams' in nativeURL) {
+				return true;
+			}
+
+			if (!('href' in nativeURL)) {
+				return false;
+			}
+		}
+	}
+	catch (error) { }
+}(this))
```

Comment:
The negative case for this detect sometimes returns false and sometimes undefined.  It could easily be consistent so I think it would benefit from that.  Currently L19-21 are basically irrelevant, no?